src/daft-distributed/src/pipeline_node/limit.rs

```diff
@@ -1,56 +1,90 @@
-use std::collections::HashMap;
+use std::{cmp::Ordering, collections::HashMap, sync::Arc};
 
+use common_daft_config::DaftExecutionConfig;
 use common_error::DaftResult;
-use common_partitioning::PartitionRef;
-use daft_local_plan::LocalPhysicalPlanRef;
+use daft_local_plan::LocalPhysicalPlan;
+use daft_logical_plan::{stats::StatsState, InMemoryInfo};
+use daft_schema::schema::SchemaRef;
+use futures::StreamExt;
 
-use super::{DistributedPipelineNode, PipelineOutput, RunningPipelineNode};
+use super::{DistributedPipelineNode, MaterializedOutput, PipelineOutput, RunningPipelineNode};
 use crate::{
-    scheduling::{scheduler::SchedulerHandle, task::SwordfishTask},
+    scheduling::{
+        scheduler::SchedulerHandle,
+        task::{SchedulingStrategy, SwordfishTask},
+    },
     stage::StageContext,
     utils::channel::{create_channel, Sender},
 };
 
 #[allow(dead_code)]
 pub(crate) struct LimitNode {
+    node_id: usize,
     limit: usize,
-    local_physical_plans: Vec<LocalPhysicalPlanRef>,
-    children: Vec<Box<dyn DistributedPipelineNode>>,
-    input_psets: HashMap<String, Vec<PartitionRef>>,
+    schema: SchemaRef,
+    config: Arc<DaftExecutionConfig>,
+    child: Box<dyn DistributedPipelineNode>,
 }
 
 impl LimitNode {
     #[allow(dead_code)]
     pub fn new(
+        node_id: usize,
         limit: usize,
-        local_physical_plans: Vec<LocalPhysicalPlanRef>,
-        children: Vec<Box<dyn DistributedPipelineNode>>,
-        input_psets: HashMap<String, Vec<PartitionRef>>,
+        schema: SchemaRef,
+        config: Arc<DaftExecutionConfig>,
+        child: Box<dyn DistributedPipelineNode>,
     ) -> Self {
-        // We cannot have empty local physical plans
-        assert!(!local_physical_plans.is_empty());
-        // If we have children, we must have input psets, and we must have a single local physical plan
-        if !children.is_empty() {
-            assert!(input_psets.is_empty());
-            assert!(local_physical_plans.len() == 1);
-        }
         Self {
+            node_id,
             limit,
-            local_physical_plans,
-            children,
-            input_psets,
+            schema,
+            config,
+            child,
         }
     }
 
-    #[allow(dead_code)]
     async fn execution_loop(
-        _scheduler_handle: SchedulerHandle<SwordfishTask>,
-        _local_physical_plans: Vec<LocalPhysicalPlanRef>,
-        _input_node: Option<RunningPipelineNode>,
-        _input_psets: HashMap<String, Vec<PartitionRef>>,
-        _result_tx: Sender<PipelineOutput>,
+        input: RunningPipelineNode,
```

Comment:
A bunch of these types were Generic over `SwordfishTask` in the original PR (https://github.com/Eventual-Inc/Daft/pull/4326/files#diff-9bed41e1d6ab627e02d2708cf9e22249f394d7de14c7ce9c1dcff89806fdb018R107), but it seemed unnecessary for this PR since it wasn't generic over anything else. So I kept it like normal